src/simulation/SimulationManager.java

```diff
@@ -172,13 +178,21 @@ public boolean performStep(int steps, List<String> constraints) {
             return false;
         }
 
-        CompModule compModule = AlloyInterface.compile(alloyModelFile.getAbsolutePath());
-        if (compModule == null) {
-            System.out.println("Cannot perform step. Could not parse model.");
+        CompModule compModule = null;
+        try {
+            compModule = AlloyInterface.compile(alloyModelFile.getAbsolutePath());
+        } catch (Err e) {
+            System.out.println("Cannot perform step. Internal error.");
             return false;
         }
 
-        A4Solution sol = AlloyInterface.run(compModule);
+        A4Solution sol = null;
+        try {
+            sol = AlloyInterface.run(compModule);
+        } catch (Err e) {
+            System.out.println("Cannot perform step. Internal error.");
+            return false;
+        }
         if (sol == null || !sol.satisfiable()) {
```

Comment:
Do we still need to null-check `sol`?

Reply:
I don't think we have any situations where the input CompModule has no commands, so I think the answer is no.